### PR TITLE
Support completion for --cluster and --user

### PR DESCRIPTION
**Pull request by superbrothers**
**What this PR does / why we need it**:

kubectl is supported completion for `--cluster` and `--user`.

```
$ kubectl --user=<tab>
--user=minikube  --user=user01

$ kubectl --cluster=<tab>
--cluster=cluster01  --cluster=minikube
```

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add completion support for --cluster and --user to kubectl
```

### CODE REVIEW

**Comment by adohe** [comment]
@k8s-bot ok to test

**Comment by superbrothers** [comment]
As far as I see the above test, it seems that this failure does not related to this PR.

**Comment by adohe** [comment]
@k8s-bot verify test this

**Comment by superbrothers** [comment]
What state does this PR? Should I do something?

**Comment by soltysh** [comment]
@superbrothers this looks ok, mind updating the release notes in the initial comment so that it reflects the newly added feature?

**Comment by superbrothers** [comment]
@soltysh Thank you for your comment. I updated the release note in the initial comment.

**Comment by soltysh** [comment]
/lgtm

You need to wait for Fabiano or TonyAdo to approve it.

**Comment by superbrothers** [comment]
Umm... It seems that e2e testing failed again where is unrelated to this PR.

**Comment by soltysh** [comment]
@k8s-bot cvm gce e2e test this

File: pkg/kubectl/cmd/cmd.go [hunk 1]
```diff
@@ -76,10 +76,26 @@ __kubectl_get_namespaces()
...
+    __kubectl_parse_config "clusters"
+}
+
+__kubectl_config_get_users()
+{
+    __kubectl_parse_config "users"
+}
+
+# $1 has to be "contexts", "clusters" or "users"
+__kubectl_config_get()
```
Comment (janetkuo) [inline pkg/kubectl/cmd/cmd.go]: Should be `__kubectl_parse_config()`

Comment (janetkuo) [inline pkg/kubectl/cmd/cmd.go]: This breaks #29386

**Pull request merged**